test_runner/src/main/kotlin/ftl/reports/outcome/BillableMinutes.kt

```diff
@@ -10,33 +11,31 @@ data class BillableMinutes(
     val physical: Long = 0
 )
 
-fun List<Step>.calculateAndroidBillableMinutes(
-    projectId: String,
-    timeoutValue: Long
-): BillableMinutes =
-    groupByDeviceType(projectId).run {
+fun List<Step>.calculateAndroidBillableMinutes(projectId: String, timeoutValue: Long): BillableMinutes = this
+    .groupBy { deviceType(it.deviceModel(), projectId) }
+    .run {
         BillableMinutes(
-            virtual = get(true)?.sumBillableMinutes(timeoutValue) ?: 0,
-            physical = get(false)?.sumBillableMinutes(timeoutValue) ?: 0
+            virtual = get(DeviceType.VIRTUAL)?.sumBillableMinutes(timeoutValue) ?: 0,
```

Comment:
What do you think of introducing an extension function for cases like this eg 
virtual = get(DeviceType.VIRTUAL)?.sumBillableMinutes(timeoutValue) ?: 0,
```suggestion
            virtual = get(DeviceType.VIRTUAL)?.sumBillableMinutes(timeoutValue).orZero(),
```
We already make use of it fairly extensively in other ways such as `orUnknown` `orNull` etc

Reply:
I think it won't work with this particular case. `sumBillableMinutes` never returns `null` so we would need to move `null` through rewritten `sumBillableMinutes` sth like
```kotlin
private fun List<Step>?.sumBillableMinutes(timeout: Long) = this?.run {
    mapNotNull { it.getBillableSeconds(default = timeout) }
    .map { billableMinutes(it) }
    .sum()
}
```
which is now having additional logic (null check)
Then it should (IMO) be renamed to `sumBillableMinutesOrNull`

I like the idea but I am not convinced if this is good place to implement it.
On the other hand this would follow the convention that we have (as you mentioned) in the project.
I don't have strong opinion about it, let me know what do you think.